falcon/routing/converters.py

```diff
@@ -59,6 +65,28 @@ def convert(self, fragment):
         return value
 
 
+class DateTimeConverter(object):
+    """Converts a field value to a datetime.
+
+    Keyword Args:
+        format_string (str): String used to parse the param value
+            into a datetime. Any format recognized by strptime() is
+            supported (default ``'%Y-%m-%dT%H:%M:%SZ'``).
+    """
+
+    __slots__ = ('_format_string', '_strptime')
+
+    def __init__(self, format_string='%Y-%m-%dT%H:%M:%SZ'):
+        self._format_string = format_string
+
+    def convert(self, fragment):
+        try:
+            return strptime(fragment, self._format_string)
+        except ValueError:
+            return None
+
+
 BUILTIN = (
     ('int', IntConverter),
+    ('dt', DateTimeConverter),
```

Comment:
> There is no explicit objection to having this converter but what is the purpose of adding this? And where did the need come from? Only asking to understand requirements

Timestamps are sometimes used in URLs to group resources chronologically. I'm also thinking ahead to when we will add support for query string parameters in the URI templates, in which case this converter could be used in lieu of `req.get_param_as_datetime()`.

> Even though you provided format_string as a keyword argument, a user that may need to provide a different format will have to create a customer class (maybe by inheriting from DateTimeConverter) to provide a format of their choice. Is this ok or expected for usage?

This should become more clear once I add the documentation (in a future PR), but parameters can be passed into the URI template field expressions, obviating the need to subclass. E.g.,:

```py
api.add_route('/{day:dt("%Y-%m-%d")/messages', messages_resource)
```